### PR TITLE
Improve build process

### LINKED ISSUE #87: New build system
The webui is about to get an overhaul, and the build system needs to be figured out before too much work is done.

Here are some goals

 - [ ] Modularize everything. See #73 
 - [ ] Create test suit. See #38 
 - [ ] Build ES2015 with Babel

The two main options are Webpack and Browserify.

## Browserify:

**Pros:**

 - Simple, follows Unix philosophy
 - Mature and fast

**Cons:**

 - Integrating non-JS assets requires writing non portable shell scripts

## Webpack:

**Pros:**

 - Fully featured. Plugins allow for awesome things like react hot-reloading
 - Loads most any type of file. All assets can be processed with one tool

**Cons:**

 - A bit of magic. Complicated config(well documented though)

### LINKED ISSUE #139: Bring back the log view
* `log/tail` works in the new js-ipfs-api
* To ensure we are not running out of memory the log should 
  * be streamed into a ring buffer with a max of 1000 lines
  * stopped when navigating away from the log view and the ring buffer cleaned up from memory

### PR DESCRIPTION
**Pull request by Dignifiedquire**
- [x] Handle everything through webpack
- [x] Hot reload
- [x] No more `make`
- [x] Lint happy
- [x] Test build step
- [x] Setup tests
- [ ] ~~Figure out how to connect to a local daemon for testing~~
- [x] Move build directory into `dist`
- [x] Ensure relative paths are working in the resulting bundle (ref https://github.com/HenrikJoreteg/hjs-webpack/issues/119)
- [x] Bring back logging, closes #139 
- [x] Convert to es2015 module system

Closes #87

Depends on https://github.com/ipfs/js-ipfs-api/pull/177

### CODE REVIEW

**Comment by Dignifiedquire** [comment]
ping @diasdavid for some good CR

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 node_js:
   - stable
 
+before_install:
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+
 script:
   - npm run lint
+  - npm test
```
Comment (diasdavid) [inline .travis.yml]: yesss! :D

File: README.md [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
 ```bash
-git clone https://github.com/ipfs/webui
-cd webui
-npm install
-# Runs server on port 3000. Set env PORT to change.
-npm run dev
+$ git clone https://github.com/ipfs/webui
+$ cd webui
+$ npm install
+# Runs server on port 3000.
```
Comment (diasdavid) [inline README.md]: The env variable doesn't work anymore?

File: README.md [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
+$ npm start
 ```
 
-This will open a browser at <http://localhost:3000>
+## Building
+
+```bash
+$ npm run build
+# The result will be in /public
```
Comment (diasdavid) [inline README.md]: don't want to bikeshed here, but since we have been called the folder compiled/bundled things `dist` across the other projects, can we keep that name?

File: app/scripts/linkhandler.js [hunk 4]
```diff
@@ -1,4 +1,3 @@
-'use strict'
 // var Router = require('react-router')
 var $ = require('jquery')
```
Comment (diasdavid) [inline app/scripts/linkhandler.js]: removing jquery will be for a next PR? :) //cc @RichardLitt

File: README.md (hunk 2 above)
Comment (Dignifiedquire) [inline README.md]: no

File: README.md (hunk 3 above)
Comment (Dignifiedquire) [inline README.md]: I don't mind

File: app/scripts/linkhandler.js (hunk 4 above)
Comment (Dignifiedquire) [inline app/scripts/linkhandler.js]: yep already an issue open for that, but requires some rewriting of components, as there are very horrible things happening there with jquery..

File: app/scripts/pages/config.js [hunk 5]
```diff
@@ -9,6 +9,8 @@ var Config = React.createClass({
   getInitialState: function () {
     var t = this
     t.props.ipfs.config.show(function (err, configStream) {
+      if (err) console.log(err)
```
Comment (diasdavid) [inline app/scripts/pages/config.js]: no return statement here? 

btw, we can definitely have more logs for debug mode in browser things we do. I really like [bows](https://www.npmjs.com/package/bows) for logging in browser apps, do you have any that you like more?

File: app/scripts/pages/config.js [hunk 6]
```diff
@@ -9,6 +9,8 @@ var Config = React.createClass({
...
+
       var config = ''
       configStream.on('data', function (chunk) {
```
Comment (diasdavid) [inline app/scripts/pages/config.js]: I way to make this nicer/simpler (and for any case we have a readable stream) is piping it to a writable stream and one that works really well is https://www.npmjs.com/package/bl so that:

configStream.pipe(bl(function (err, data) {})

This way, we don't have to do the chunking glue ourselves every time and we also don't forget to check for the error on the readable stream.

File: app/scripts/pages/connections.js [hunk 7]
```diff
@@ -1,8 +1,7 @@
-'use strict'
 var React = require('react')
-var ConnectionList = require('../views/connectionlist.jsx')
-var Globe = require('../views/globe.jsx')
-var getLocation = require('ipfs-geoip').lookupPretty
+var ConnectionList = require('../views/connectionlist')
```
Comment (diasdavid) [inline app/scripts/pages/connections.js]: I'm curious, no `const` or arrow functions this time? Any specific reason why or are you saving that for another PR?

File: app/scripts/pages/config.js (hunk 5 above)
Comment (Dignifiedquire) [inline app/scripts/pages/config.js]: This is mostly to make the linter happy tbh. There is no real error handling anywhere atm. 

Bows looks nice but what's really important is displaying errors to users in a friendly format, which we need to tackle soon

File: app/scripts/pages/config.js (hunk 6 above)
Comment (Dignifiedquire) [inline app/scripts/pages/config.js]: bl looks really nice :)

File: app/scripts/pages/connections.js (hunk 7 above)
Comment (Dignifiedquire) [inline app/scripts/pages/connections.js]: saving that for later, there is so much to do here that I'm trying to do one step at a time :)

File: app/scripts/pages/logs.js [hunk 8]
```diff
@@ -11,7 +10,7 @@ var Logs = React.createClass({
     host: React.PropTypes.string
   },
   getInitialState: function () {
-    var t = this
+    // var t = this
     var req = null
     // TODO: fix this
     // var req = this.props.ipfs.log.tail(function (err, stream) {
```
Comment (diasdavid) [inline app/scripts/pages/logs.js]: tail now works through js-ipfs-api! one more thing to add to the webui as a feature //cc @RichardLitt :)

File: app/scripts/utils/localStorage.js [hunk 9]
```diff
@@ -31,11 +30,11 @@ function doOperation (op) {
     if (op === 'getItem') {
       try {
         result = JSON.parse(result)
-      } catch(e) {}
+      } catch (e) {}
```
Comment (diasdavid) [inline app/scripts/utils/localStorage.js]: let's log it at least :)

Comment (Dignifiedquire) [inline app/scripts/utils/localStorage.js]: in another pr, trying not to change the components unless I have to this round (just linting here)

File: app/scripts/views/page.js [hunk 10]
```diff
@@ -98,17 +103,17 @@ var Page = React.createClass({
                         <ul className='nav navbar-nav navbar-right collapse navbar-collapse'>
                           <li>
                               <a href='http://ipfs.io' target='_blank' data-toggle='tooltip' data-placement='bottom' title={i18n.t('About IPFS')}>
-                                  <img src='./static/img/help.png' alt='Help' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Help')}</span>
+                                  <img src={require('../../img/help.png')} alt='Help' className='img-responsive icon'/><span className='sr-only'>{i18n.t('Help')}</span>
```
Comment (diasdavid) [inline app/scripts/views/page.js]: wait, we can require .png now? woooo!

Comment (Dignifiedquire) [inline app/scripts/views/page.js]: it's magic :rainbow:

File: app/styles/base.less [hunk 11]
```diff
@@ -347,7 +347,7 @@ thead {
   width: 22px;
   height: 22px;
   display: inline-block;
-  background: url(img/log.png) no-repeat;
+  background: url(../img/log.png) no-repeat;
```
Comment (diasdavid) [inline app/styles/base.less]: On relative urls, have you tried in both `localhost:5001/webui` and `localhost:port/ipfs/Qmhash` stuff and so on? Just to make sure the imports don't get messed

File: package.json [hunk 12]
```diff
@@ -3,46 +3,69 @@
...
+    "postcss-loader": "^0.8.0",
     "pre-commit": "^1.1.2",
-    "proxy-middleware": "^0.15.0",
-    "standard": "5.3.1"
+    "react-addons-test-utils": "^0.14.3",
+    "stream-http": "^2.0.2",
+    "style-loader": "^0.13.0",
+    "url-loader": "^0.5.7",
+    "webpack": "^1.12.9",
+    "webpack-dev-server": "^1.14.0"
```
Comment (diasdavid) [inline package.json]: I think it is time to create "@dignifiedquire's npm bundle" and just install all of karma, mocha, eslint, babel and webpack with one dependency :D

Comment (Dignifiedquire) [inline package.json]: I'm actually considering doing sth like that, gets painful doing the same things everywhere over and over again..

File: versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js [hunk 13]
```diff
@@ -3690,7 +3690,7 @@ http.request = function (params, cb) {
         params.host = params.host.split(':')[0];
     }
     if (!params.port) params.port = params.protocol == 'https:' ? 443 : 80;
-    
+
     var req = new Request(new xhrHttp, params);
     if (cb) req.on('response', cb);
     return req;
```
Comment (diasdavid) [inline versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js]: this is weird. `if` with two statements and no block followed by an else? 

(plus semicolons?)

File: app/styles/base.less (hunk 11 above)
Comment (Dignifiedquire) [inline app/styles/base.less]: need to check but webpack rewrites all urls so I'm pretty sure it will work

File: versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js [hunk 14]
```diff
@@ -3811,23 +3811,23 @@ var Request = module.exports = function (xhr, params) {
...
         + (params.path || '/')
     ;
-    
+
     if (typeof params.withCredentials === 'undefined') {
         params.withCredentials = true;
     }
 
     try { xhr.withCredentials = params.withCredentials }
     catch (e) {}
```
Comment (diasdavid) [inline versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js]: at least, lets log it

File: versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js [hunk 15]
```diff
@@ -3811,23 +3811,23 @@ var Request = module.exports = function (xhr, params) {
...
-    
+
     if (params.responseType) try { xhr.responseType = params.responseType }
     catch (e) {}
```
Comment (diasdavid) [inline versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js]: same here

File: versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js [hunk 16]
```diff
@@ -4092,7 +4092,7 @@ Response.prototype.handle = function (res) {
             }
         }
         catch (err) {}
```
Comment (diasdavid) [inline versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js]: same here

File: versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js [hunk 17]
```diff
@@ -11248,10 +11248,10 @@ var React = require('react');
  *     mixins: [ Router.State ],
```
Comment (diasdavid) [inline versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js]: these comments can be deleted, right? no need for them, just boilerplate

File: versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js (hunk 13 above)
Comment (Dignifiedquire) [inline versions/QmaaqrHyAQm7gALkRW8DcfGX3u8q9rWKnxEMmf7m9z515w/static/bundle.min.js]: this is a built file, ignore it, not sure that ended up in here..

**Comment by diasdavid** [comment]
Hi @Dignifiedquire, just finished the CR, just a couple of small things here and there but overall LGTM :) thank you!

@RichardLitt you might be interested in CR'ing and checking the progress @Dignifiedquire made here at the webui

**Comment by Dignifiedquire** [comment]
Thanks will clean those things and then get if merged. Also everyone go write tests for the webui no excuses anymore :P

**Comment by RichardLitt** [comment]
Will CR this. Have it as a To do for tomorrow, if that's OK.

File: app/scripts/pages/config.js (hunk 5 above)
Comment (harlantwood) [inline app/scripts/pages/config.js]: +1 for bows

File: package.json [hunk 18]
```diff
@@ -3,46 +3,69 @@
...
-    "react-dom": "0.14.2",
-    "react-localstorage": "0.2.8",
-    "react-router": "0.13.5",
-    "three": "0.73.0"
+    "bootstrap": "^3.3.5",
+    "d3": "^3.5.6",
+    "debug": "^2.2.0",
+    "font-awesome": "^4.5.0",
+    "i18next": "^2.0.5",
+    "i18next-browser-languagedetector": "0.0.11",
```
Comment (harlantwood) [inline package.json]: Why lock this and following libs to exact version instead of using ^version?

**Comment by harlantwood** [comment]
This looks great, nice work.

File: app/styles/base.less (hunk 11 above)
Comment (Dignifiedquire) [inline app/styles/base.less]: Looks like it's not working as I'd hope to: https://github.com/HenrikJoreteg/hjs-webpack/issues/119 will need to fix this before we can merge

File: package.json (hunk 18 above)
Comment (Dignifiedquire) [inline package.json]: No specific reason, not sure how that happened :confused:

File: README.md [hunk 19]
```diff
@@ -1,18 +1,24 @@
...
 
 [![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io) [![](https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square)](http://ipfs.io/) [![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs) [![Dependency Status](https://david-dm.org/ipfs/webui.svg?style=flat-square)](https://david-dm.org/ipfs/webui) [![Circle CI](https://img.shields.io/circleci/project/ipfs/webui/master.svg?style=flat-square)](https://circleci.com/gh/ipfs/webui) [![Travis CI](https://img.shields.io/travis/ipfs/webui/master.svg?style=flat-square)](https://travis-ci.org/ipfs/webui)
 
-> IPFS Web Interaface
+> The web interface for [IPFS](https://ipfs.io/)
 
-## Dev
+## Development
+
+Make sure [node.js](https://nodejs.org/) and [npm](https://docs.npmjs.com/) are installed and in your path.
```
Comment (RichardLitt) [inline README.md]: Add 'Make sure you are running the daemon in another window'

File: .gitignore [hunk 20]
```diff
@@ -1,7 +1,5 @@
-build/
```
Comment (RichardLitt) [inline .gitignore]: Why did you remove build?

File: .travis.yml [hunk 21]
```diff
@@ -1,9 +1,13 @@
-# See http://docs.travis-ci.com/user/workers/container-based-infrastructure/
```
Comment (RichardLitt) [inline .travis.yml]: Why remove this?

File: README.md [hunk 22]
```diff
@@ -1,18 +1,24 @@
-IPFS webui
-==========
+# IPFS webui
 
 [![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io) [![](https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square)](http://ipfs.io/) [![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs) [![Dependency Status](https://david-dm.org/ipfs/webui.svg?style=flat-square)](https://david-dm.org/ipfs/webui) [![Circle CI](https://img.shields.io/circleci/project/ipfs/webui/master.svg?style=flat-square)](https://circleci.com/gh/ipfs/webui) [![Travis CI](https://img.shields.io/travis/ipfs/webui/master.svg?style=flat-square)](https://travis-ci.org/ipfs/webui)
```
Comment (RichardLitt) [inline README.md]: Add the Contributing banner in here. Or at least a Contribute section linking to the contrib file.

File: README.md [hunk 23]
```diff
@@ -1,18 +1,24 @@
...
+Make sure [node.js](https://nodejs.org/) and [npm](https://docs.npmjs.com/) are installed and in your path.
 
 ```bash
-git clone https://github.com/ipfs/webui
-cd webui
-npm install
-# Runs server on port 3000. Set env PORT to change.
-npm run dev
+$ git clone https://github.com/ipfs/webui
+$ cd webui
```
Comment (RichardLitt) [inline README.md]: These two lines are kind of unnecessary. I renamed mine to `ifps-webui`, and I suspect others do the same.

File: app/scripts/linkhandler.js (hunk 4 above)
Comment (RichardLitt) [inline app/scripts/linkhandler.js]: Woot.

File: app/scripts/include/globe.js [hunk 24]
```diff
@@ -1,4 +1,3 @@
-'use strict'
```
Comment (RichardLitt) [inline app/scripts/include/globe.js]: Why did you remove this repeatedly? curious.

File: app/scripts/pages/connections.js (hunk 7 above)
Comment (RichardLitt) [inline app/scripts/pages/connections.js]: What's the advantage of doing them later? Are they time-intensive to use?

File: app/scripts/pages/connections.js [hunk 25]
```diff
@@ -42,28 +41,28 @@ var Connections = React.createClass({
...
+          //       peer.location = res
+          //       t.state.locations[peer.ID] = res
+          //       t.setState({
+          //         peers: peers,
+          //         locations: t.state.locations,
+          //         nonce: t.state.nonce++
+          //       })
+          //     })
+          //   })
+          // }
```
Comment (RichardLitt) [inline app/scripts/pages/connections.js]: Why is this commented out, and not deleted? Also; I can't see any peers on the globe; expected?

File: app/scripts/pages/logs.js (hunk 8 above)
Comment (RichardLitt) [inline app/scripts/pages/logs.js]: Nice!

File: app/scripts/pages/logs.js [hunk 26]
```diff
@@ -11,7 +10,7 @@ var Logs = React.createClass({
     host: React.PropTypes.string
   },
   getInitialState: function () {
-    var t = this
+    // var t = this
     var req = null
     // TODO: fix this
```
Comment (RichardLitt) [inline app/scripts/pages/logs.js]: Should we be going through and taking out these low-level To Do's into issues?

File: app/scripts/utils/__tests__/path-test.js [hunk 27]
```diff
@@ -0,0 +1,10 @@
+/* eslint-env mocha */
```
Comment (RichardLitt) [inline app/scripts/utils/__tests__/path-test.js]: Why the naming as __tests__?

File: app/scripts/views/page.js [hunk 28]
```diff
@@ -83,7 +88,7 @@ var Page = React.createClass({
                     <div className='row'>
                           <div className='navbar-header'>
                             <Link className='navbar-brand col-xs-12' to='home'>
-                              <img src='./static/img/logo.png' alt='IPFS' className='img-responsive logo'/><span className='sr-only'>{i18n.t('IPFS')}</span>
+                              <img src={require('../../img/logo.png')} alt='IPFS' className='img-responsive logo'/><span className='sr-only'>{i18n.t('IPFS')}</span>
```
Comment (RichardLitt) [inline app/scripts/views/page.js]: This is much less pretty. Can we do something about ../../?

Comment (RichardLitt) [inline app/scripts/views/page.js]: Also, should we really be translating IPFS?

File: app/styles/bundle.less [hunk 29]
```diff
@@ -1,9 +1,6 @@
-/* use paths from root folder */
-@import "node_modules/bootstrap/less/bootstrap.less";
-@import "less/base.less";
-@import "node_modules/font-awesome/less/font-awesome.less";
-
-@fa-font-path: "node_modules/font-awesome/fonts";
+@import "../../node_modules/bootstrap/less/bootstrap.less";
+@import "./base.less";
+@import "../../node_modules/font-awesome/less/font-awesome.less";
```
Comment (RichardLitt) [inline app/styles/bundle.less]: Why aren't we using Sass?

File: package.json (hunk 12 above)
Comment (RichardLitt) [inline package.json]: Just do it. Never do the same thing more than twice if there's a way not to.

File: html/index.html [hunk 30]
```diff
@@ -1,16 +0,0 @@
-<!DOCTYPE html>
```
Comment (RichardLitt) [inline html/index.html]: Where'd the old file go?

File: app/styles/bundle.less (hunk 29 above)
Comment (Dignifiedquire) [inline app/styles/bundle.less]: nope, it was less before and I didn't change that

File: html/index.html (hunk 30 above)
Comment (Dignifiedquire) [inline html/index.html]: Magic ;) It get's generated by webpack now :)

File: app/scripts/views/page.js (hunk 28 above)
Comment (Dignifiedquire) [inline app/scripts/views/page.js]: > This is much less pretty. Can we do something about ../../?

Yes and no, in the long run we need an image component like this: `<Image src="logo.png" />` that handles these things for us. But for now it's just getting things working with the new setup

File: app/scripts/pages/logs.js (hunk 26 above)
Comment (Dignifiedquire) [inline app/scripts/pages/logs.js]: Feel free to do so

Comment (Dignifiedquire) [inline app/scripts/pages/logs.js]: There is just so much to do when I even look at a single component in here that I could create 3 issues from that alone..so I haven't started with that

File: app/scripts/utils/__tests__/path-test.js (hunk 27 above)
Comment (Dignifiedquire) [inline app/scripts/utils/__tests__/path-test.js]: react convention, not a 100% sure I want to go with it though

**Comment by RichardLitt** [comment]
So, I ran `npm run lint`, and I got: `✖ 184769 problems (184769 errors, 0 warnings)`.

Uhm.....

**Comment by Dignifiedquire** [comment]
Delete the old `build` folder

**Comment by RichardLitt** [comment]
And no errors. Sweet. Thanks! 

Looks good to me. The code seems pretty solid, but I can't speak as to it's functionality much at the moment.

File: app/scripts/include/globe.js (hunk 24 above)
Comment (Dignifiedquire) [inline app/scripts/include/globe.js]: Because es2015 modules are in strict mode by default, so when we use babel to transpile all our code is strict, without us having to write it everywhere :)

File: .gitignore (hunk 20 above)
Comment (Dignifiedquire) [inline .gitignore]: because it's now called `dist`

File: app/scripts/include/globe.js (hunk 24 above)
Comment (RichardLitt) [inline app/scripts/include/globe.js]: Sweet.

File: .travis.yml (hunk 21 above)
Comment (Dignifiedquire) [inline .travis.yml]: Seems superfluous to me

File: app/scripts/utils/__tests__/path-test.js (hunk 27 above)
Comment (Dignifiedquire) [inline app/scripts/utils/__tests__/path-test.js]: Thoughts @diasdavid @RichardLitt is this good, or should we move all tests to a top level `tests` folder?

**Comment by Dignifiedquire** [comment]
Thanks to @travisperson and @RichardLitt we are not pretty close to merging this :)

**Comment by RichardLitt** [comment]
@Dignifiedquire What else do you need help with?

**Comment by Dignifiedquire** [comment]
@RichardLitt the last thing before I merge this is deciding on where to put the tests

**Comment by Dignifiedquire** [comment]
@RichardLitt after that, all the other issues with `help wanted` tags :)

**Comment by RichardLitt** [comment]
Cool. I'd suggest making at least one test before merging, as an example. Then we can start throwing more in.

**Comment by Dignifiedquire** [comment]
@RichardLitt there is one test: https://github.com/ipfs/webui/blob/webpack/app/scripts/utils/__tests__/path-test.js

File: app/scripts/utils/__tests__/path-test.js (hunk 27 above)
Comment (harlantwood) [inline app/scripts/utils/__tests__/path-test.js]: +1 for top level tests folder.  That's how most projects, and most of our projects do it I believe.

**Comment by Dignifiedquire** [comment]
Happy New Year 🎉

**Comment by diasdavid** [comment]
😀😀😀😀😀😀😀

File: app/styles/base.less (hunk 11 above)
Comment (RichardLitt) [inline app/styles/base.less]: Was this fixed? Still an outstanding comment.

Comment (Dignifiedquire) [inline app/styles/base.less]: Look at the issue ;)

**Comment by RichardLitt** [comment]
Wooo!

File: app/styles/base.less (hunk 11 above)
Comment (Dignifiedquire) [inline app/styles/base.less]: also https://github.com/ipfs/webui/pull/148
